staging/src/k8s.io/client-go/tools/events/event_recorder.go

```diff
@@ -49,9 +49,12 @@ func (recorder *recorderImpl) Eventf(regarding runtime.Object, related runtime.O
 		klog.Errorf("Could not construct reference to: '%#v' due to: '%v'. Will not report event: '%v' '%v' '%v'", regarding, err, eventtype, reason, message)
 		return
 	}
-	refRelated, err := reference.GetReference(recorder.scheme, related)
-	if err != nil {
-		klog.Errorf("Could not construct reference to: '%#v' due to: '%v'.", related, err)
+	var refRelated *v1.ObjectReference
+	if related != nil {
```

Comment:
right now I'm making it just tolerate the `nil` case. Please let me know if you think we should tolerate all cases.

Reply:
We should just switch to Infof as log level

Reply:
I would even say Infof.V(3) or sth...

Reply:
thinking more about this, I'd put a `V(9)` as this would be used only for debug

Reply:
I'm changing the logic to `V(9)`. Please take another look.